### PR TITLE
[FS-1489] [FS-1776] allow higher versions of python, fix dotenv bug, more verbosity

### PR DESCRIPTION
**Pull request by martukas**
Addresses [FS-1489](https://elroyair.atlassian.net/browse/FS-1489) and [FS-1776](https://elroyair.atlassian.net/browse/FS-1776)

* allows versions of python above 3.11
* upgrades poetry packages
* bumps pre-commit hook versions
* More verbose output, particularly re. JIRA validation with backtrace
* Fixes `dotenv` bug where it searches parent paths of installed library, and not that of the repo which has initiated the script
* Prints ticket summary, if found

[FS-1489]: https://elroyair.atlassian.net/browse/FS-1489?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FS-1776]: https://elroyair.atlassian.net/browse/FS-1776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by elroyair-tanner-borglum — APPROVED** [review]
This seems straightforward and looks good to me!

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@ __pycache__
 # Local testing
 .env
 dummyfile
+
+# prettier-generated junk
+node_modules
```
Comment (karanchawla) [inline .gitignore:15]: :mild-panic-intensifies:

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,37 +15,23 @@ repos:
       - id: mixed-line-ending # replaces or checks mixed line ending.
       - id: requirements-txt-fixer # sorts entries in requirements.txt.
 
-  - repo: https://github.com/psf/black
-    rev: "23.11.0"
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.4.9
```
Comment (karanchawla) [inline .pre-commit-config.yaml:20]: :pinched_fingers: hell yeah!

**Review by karanchawla — APPROVED** [review]
LGTM

**Pull request merged**